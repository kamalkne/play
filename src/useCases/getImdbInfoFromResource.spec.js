const getImdbInfoFromResourceFactory = require('./getImdbInfoFromResource')
const mockAllGateways = require('../../testHelpers')
const movieDetails = require('../../testHelpers/mocks/movieDetails.json')

describe('getImdbInfoFromResource', () => {
    let getImdbInfoFromResource
    const gateways = mockAllGateways()

    beforeEach(() => {
        getImdbInfoFromResource = getImdbInfoFromResourceFactory({ gateways })
    })

    describe('Get Imdb Info',() => {
        beforeEach(() => {
            gateways.movieGateway.getMovieDetails.mockResolvedValueOnce(movieDetails)
        })
        test('should return IMDB data',async () => {
            expect(await getImdbInfoFromResource('resourceLink')).toMatchSnapshot()
        })
    })

    describe('Error',() => {
        beforeEach(() => {
            gateways.movieGateway.getMovieDetails.mockResolvedValueOnce(undefined)
        })
        test('should return undefined when movie details are not available',async () => {
            expect(await getImdbInfoFromResource('resourceLink')).toBeUndefined()
        })
    })
})
