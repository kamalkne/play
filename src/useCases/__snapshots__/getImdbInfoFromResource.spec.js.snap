// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getImdbInfoFromResource Get Imdb Info should return IMDB data 1`] = `
Object {
  "id": "tt2543164",
  "rating": "7.9",
  "url": "http://www.imdb.com/title/tt2543164?ref_ext_viaplay",
  "votes": "631 829",
}
`;
